hack/make-and-check-size

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+#
+# make-and-check-size - wrapper around 'make' that also checks binary growth
+#
+# This script is intended to be run via 'git rebase -x', in a Makefile rule
+# such as:
+#
+#    build-all-new-commits:
+#       CONTEXT_DIR=$(shell mktemp -d --tmpdir make-size-check.XXXXXXX); \
+#       git rebase $(GIT_BASE_BRANCH)^ -x "hack/make-and-check-size $$CONTEXT_DIR"; \
+#       $(RM) -rf $$CONTEXT_DIR
+#
+# ...which has long been a part of our usual CI, one that makes sure that
+# each commit (in a multi-commit PR) can be compiled individually. By
+# adding the '^' to GIT_BASE_BRANCH we establish a baseline and store
+# the binary sizes of each file (podman, podman-remote) prior to our PR.
+#
+# CONTEXT_DIR is a temporary directory used to store the original sizes
+# of each binary file under bin/
+#
+# *IMPORTANT NOTE*: this script will leave the git checkout in a funky state!
+# (because we rebase onto a nonterminal commit). I believe this is OK, since
+# this makefile target is used only in CI and only in a scratch VM. Running
+# this in a development environment would yield unpredictable results anyway,
+# by rebasing onto origin/main by default and by leaving an aborted rebase
+# on failure.
+#
+ME=$(basename $0)
+
+###############################################################################
+# BEGIN end-user-customizable settings
+
+# Maximum allowable size, in bytes
+MAX_BIN_GROWTH=$((50 * 1024))
+
+# Github label which allows overriding this check
+OVERRIDE_LABEL=bloat_approved
+
+# END   end-user-customizable settings
+###############################################################################
+
+#
+# Helper function: queries github for labels on this PR
+#
+function bloat_approved() {
+    # Argument is the actual size increase in this build.
+    # FIXME: 2022-03-21: this is not actually used atm, but Ed hopes some day
+    #        to implement a more robust size-override mechanism, such as by
+    #        requiring a MAX_BIN_GROWTH=nnn statement in github comments.
+    local actual_growth="$1"
+
+    if [[ -z "$CIRRUS_PR" ]]; then
+        echo "$ME: cannot query github: \$CIRRUS_PR is undefined" >&2
+        return 1
+    fi
+    if [[ -z "$CIRRUS_REPO_CLONE_TOKEN" ]]; then
```

Comment:
Minor nit: It might be good to comment on `$CIRRUS_REPO_CLONE_TOKEN` that we don't really care about securing it b/c this is an open-source repo.  In case it leaks and causes someone to panic :grin: